jemalloc-sys/build.rs

```diff
@@ -54,13 +62,44 @@ fn main() {
     }
 
     fs::create_dir_all(&build_dir).unwrap();
-
-    let compiler = cc::Build::new().get_compiler();
+    // Disable -Wextra warnings - jemalloc doesn't compile free of warnings with
+    // it enabled: https://github.com/jemalloc/jemalloc/issues/1196
+    let compiler = cc::Build::new().extra_warnings(false).get_compiler();
     let cflags = compiler.args().iter().map(|s| s.to_str().unwrap())
                          .collect::<Vec<_>>().join(" ");
+    println!("CC={:?}", compiler.path());
+    println!("CFLAGS={:?}", cflags);
+
+    let jemalloc_src_dir = src_dir.join("jemalloc");
+    println!("JEMALLOC_SRC_DIR={:?}", jemalloc_src_dir);
+
+    // If the configure file does not exist, run autogen and distclean:
+    if !jemalloc_src_dir.join("configure").exists() {
+        // Run autogen:
+        let autogen = jemalloc_src_dir.join("autogen.sh");
```

Comment:
Hm I think this should generate an error, right? We don't want to publish a version of jemalloc-sys which accidentally doesn't have a configure script I think?

Reply:
Not checking in a configure script makes it nigh impossible to work on Windows, right? Does OSX or Linux have this dependency by default? I agree that it's extra work to be done but it's for a handful of people vs all uses of the crate

Reply:
> Does OSX or Linux have this dependency by default? 

Those are good questions, I don't know. I am on MacOSX and I have autoconf but maybe I manually installed it via homebrew (can't remember). My linux systems have it. 

> Not checking in a configure script makes it nigh impossible to work on Windows, right? 

That's a good question. I don't know if running `autoconf.sh` works on Windows, I am hopeful, but it might require mingw or something. We'll see how that goes with the appveyor CI.

Otherwise I would prefer to put the configure script in a different directory, and "copy" it manually to the jemalloc directory as part of the `build.rs` over using a fork.

Reply:
Sure yeah that seems reasonable to me (copying in the configure script)

Reply:
Yeah, that way we can run `autogen.sh` on Linux CI and detect whether the `configure` script changes due to upstream changes to `autogen.sh` (like it happened from jemalloc 4.5 to 5.0).